src/pull-request.ts

```diff
@@ -1,23 +1,35 @@
-import * as sourcegraph from 'sourcegraph'
+import { app, commands, configuration, ExtensionContext, search, NotificationType } from 'sourcegraph'
+import { runCodemodQuery } from './codemod'
 import { issuePullRequestWithToken } from './pull-request-core'
+import { bundlePullRequests } from './user-input-sequence'
 
-/**
- * The activate function is called when one of the extensions `activateEvents`
- * conditions in package.json are satisfied.
- */
-export function activate(ctx: sourcegraph.ExtensionContext): void {
-    // const gitHubAccessToken = sourcegraph.configuration
-    //    .get()
-    //    .get('github.pr.accessToken')
-    // if (!gitHubAccessToken) {
-    //    console.log('Could not read github.pr.accessToken from configuration')
-    //    return;
-    // }
-
-    const gitHubAccessToken = 'YOUR_GH_TOKEN'
+export function activate(ctx: ExtensionContext): void {
+    // store the query string in configuration after a search. It's used to execute a GQL codemod command when we're ready to PR.
+    ctx.subscriptions.add(
+        search.registerQueryTransformer({
+            transformQuery: async query => {
+                await commands.executeCommand('updateConfiguration', 'pr.codemodQuery', query)
```

Comment:
I want to persist the latest run search query in my extension. I'm not sure where the place is for 'my app state' to live, so I store it in the user's configuration. This is probably an abuse of configuration, so where can I persist updated search queries? local storage?